tests/smartswitch/common/device_utils_dpu.py

```diff
@@ -37,29 +37,31 @@ def check_smartswitch_and_dark_mode(duthosts,
     """
 
     duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    output = duthost.os_version
+    os_version = float(re.search(r'([1-9]{1}\d*)(\.\d{0,2})', output).group())
```

Comment:
you can simply do this way "os_version = duthost.os_version.split(".")[0]" and compare directly like os_version <= "202405"

Reply:
this will not work with master branch, it's better to use skip_release same as in other tests.

Reply:
Actually, smartswitch case is little different, it can be run on os versions greater than 202405.
The check of DPUS field in platform.json is enough as well to check that. Since DPUS filed will be present in those releases.